src/utils/proposals.js

```diff
@@ -0,0 +1,366 @@
+ * @property {string} [errorMessage] Error message, present when status is failed
+ * @property {string} [oldStatus] Saga-related information, from PROPOSAL_DOWNLOAD_STATUS
+ * @property {number} [updatedAt] Saga-related last update timestamp
+ * @property {ProposalData} [data] Full decrypted proposal data
+ * @property {BackendProposalData} [rawData] Raw data received from the backend
+ */
+
+/**
+ * @typedef BackendProposalData
+ * @property {string} id
+ * @property {string} partialTx Encrypted PartialTx data
+ * @property {string} signatures Encrypted Signatures data
+ * @property {number} timestamp Latest update timestamp
+ * @property {{partialTx: string, timestamp: number}[]} history Historic data
+ */
+
+/**
+ * Generates a random positive integer between the maximum and minimum values,
+ * with the default minimum equals zero
+ * @param {number} max
+ * @param {number} [min=0]
+ * @returns {number} Random number
+ */
+export function getRandomInt(max, min = 0) {
+    const _min = Math.ceil(min);
+    const _max = Math.floor(max);
+    return Math.floor(Math.random() * (_max - _min + 1)) + _min;
+}
+
+export const PROPOSAL_DOWNLOAD_STATUS = {
+    READY: 'ready',
+    FAILED: 'failed',
+    LOADING: 'loading',
+    INVALIDATED: 'invalidated',
+};
+
+export const PROPOSAL_SIGNATURE_STATUS = {
+    OPEN: 'Open',
+    PARTIALLY_SIGNED: 'Partially Signed',
+    SIGNED: 'Signed',
+    SENT: 'Sent',
+}
+
+
+
+/**
+ * The amount from the InputNumber component is represented in a float format, with an amount of decimal
+ * cases defined by a lib's constant. This helper function converts it back to an integer.
+ * @param {number} floatAmount
+ * @returns {number}
+ */
+export function getIntTokenAmount(floatAmount) {
+    return Math.floor(floatAmount * Math.pow(10, DECIMAL_PLACES));
+}
+
+/**
+ * Generates an empty proposal for storing on redux
+ * @param {string} password
+ * @param {HathorWallet} wallet Current wallet in use
+ * @return {{id:string, password:string, data:ProposalData}}
+ */
+export function generateEmptyProposalFromPassword(password, wallet) {
+    const partialTx = new PartialTx(wallet.getNetworkObject());
+    const pId = v4();
+
+    return {
+        id: pId,
+        password,
+        data: {
+            id: pId,
+            partialTx: partialTx.serialize(),
+            signatures: null,
+            amountTokens: 0,
+            signatureStatus: PROPOSAL_SIGNATURE_STATUS.OPEN,
+            timestamp: undefined,
+            history: []
+        },
+    };
+}
+
+/**
+ * Decodes all output scripts and identifies which elements are related to this wallet,
+ * so that exhibition is made easier.
+ *
+ * Also highlights the output address by copying its `base58` representation to the root of the `output` property.
+ * @param {PartialTx} partialTx
+ * @param {HathorWallet} wallet
+ * @param {string} [strSignatures] Optional signatures string
+ */
+export function enrichTxData(partialTx, wallet, strSignatures) {
```

Comment:
Two main reasons:
- The metadata added to the PartialTx instance here is used on other calculations throughout the `EditSwap` screen ( Ex.: `handleSaveClick`, `handleRemoveInteractions` ).
- Its calculations are quite intensive, requiring parsing scripts with the wallet's network object and serializing + deserializing transactions and signatures. I thought it better to avoid handling this potentially at render time.